modules/schedule/index.js

```diff
@@ -0,0 +1,396 @@
+              var s = Date.create(schedule);
+              if(!s.isValid())
+                return reply("Invalid relative date provided.");
+              if(!s.isFuture())
+                return reply("Cannot make reminders for past dates/times.");
+
+              // massage into a cron format
+              schedule = s.utc().format('{m} {H} {d} {M} *');
+            }
+
+            reply(newSchedule({
+              'blame': from,
+              'created': new moment(),
+              'schedule': schedule,
+              'command': command,
+              'channel': to,
+              'calls': 1,
+              'target': target
+            }));
+          },
+  schedule: {
+    _default: function(x,y,reply) {
+      reply("Usage: !schedule [<add>|<remove>|<list>|<help>] [arguments]");
+      reply("       Also see !schedule help <subcommand> for more details");
+    },
+    help: {
+      _default: null, // to be aliased later to ^
+      add: function(x,y,reply) {
+        reply("Usage: !schedule add \"timeframe\" \"command or text\"");
+        reply("       Timeframe syntax is here: http://bunkat.github.io/later/parsers.html#text");
+        reply("       If a valid command isn't specified, it is treated as text to print.");
+      },
+      remove: function(x,y,reply) {
+        reply("Usage: !schedule remove [digest]");
+        reply("       Removes the schedule specified by the digest.");
+      },
+      list: function(x,y,reply) {
+        reply("Usage: !schedule list [offset=0]");
+        reply("       Provides a list of schedules, in pages, from the specified offset of 0.");
+      },
+    },
+    add: function(r, parts, reply, command, from, to, text, raw) {
+      if(parts.length !== 2) return reply("add must have *exactly* two arguments");
+
+      reply(newSchedule({
+        'blame': from,
+        'created': new moment(),
+        'schedule': parts[0],
+        'command': parts[1],
+        'channel': to,
+        'calls': -1
+      }));
+    },
+    list: function(x, parts, reply, command, from, to) {
+      var offset = 0;
+      var count  = 5;
+      var ci     = count;
+      var oi     = 0;
+      var size = Object.keys(schedules).length;
+
+      if(parts.length >= 1)
+        oi = offset = parseInt(parts[0]);
+
+      if(offset < 0)
+        oi = offset = size + offset;
+
+      if(offset >= size)
+        return bot.sayTo(from, "There are only " + size + " schedules");
+      
+      if(size == 0)
+        return bot.sayTo(from, "There are no schedules.");
+
+      if(count+offset > size)
+        count = size-offset;
+
+      // count and pick
+      Object.keys(schedules).forEach(function(digest) {
+        if(oi > 0)
+        {
+          oi -= 1;
+          return;
+        }
+
+        if(ci == 0)
+          return;
+
+        var e = schedules[digest];
+        var channel = e.channel;
+
+        if(!channel.match(/^(#|&)/))
```

Comment:
For boolean regexes, I find `/regex/.test(string)` more readable.

Also, this properly should be something more like `if(bot.config.channelPrefixes.includes(channel[0]))` in any case rather than hardcoding the RFC.. uh.. because ... uh... nope, nevermind, I don't have any actually good reason.

Do note if you use `.includes` you'll want to run with nodejs > v4 (add the "engine" key to package.json if you do that; right now we work with anything, but I don't mind requiring modern nodejs).